core/src/main/scala/org/apache/spark/ContextCleaner.scala

```diff
@@ -188,7 +188,7 @@ private[spark] class ContextCleaner(
   private def keepCleaning(): Unit = Utils.tryOrStopSparkContext(sc) {
     while (!stopped) {
       try {
-        val reference = Option(referenceQueue.remove(ContextCleaner.REF_QUEUE_POLL_TIMEOUT))
+        val reference = Option(referenceQueue.remove())
```

Comment:
Hm, the issue is that remove() blocks indefinitely, so this would never let it check `stopped` again if nothing is in the queue?

Reply:
Hm, This is a issue too, I'm thinking, `while (!stopped)` means `while (true)`  if there is no stop all the time, the will lead to some ineffective loop, I'm not sure if there are some ineffective effects?

Reply:
+1 with @srowen -- it seems to me that the current behavior is intentional, to allow for periodic checking of `stopped` to allow for the loop to exit.